v2/i18n/icons_test.go

```diff
@@ -0,0 +1,190 @@
+	},
+	{
+		tag:                 language.Polish, // pl
+		appleStoreLink:      "https://cdn.ag5.com/mail-assets/mobile/appstore/pl.png",
+		googlePlayStoreLink: "https://cdn.ag5.com/mail-assets/mobile/googleplay/pl.png",
+	},
+	{
+		tag:                 language.Portuguese, // pt
+		appleStoreLink:      "https://cdn.ag5.com/mail-assets/mobile/appstore/pt.png",
+		googlePlayStoreLink: "https://cdn.ag5.com/mail-assets/mobile/googleplay/pt.png",
+	},
+	{
+		tag:                 language.BrazilianPortuguese, // pt-BR
+		appleStoreLink:      "https://cdn.ag5.com/mail-assets/mobile/appstore/pt-br.png",
+		googlePlayStoreLink: "https://cdn.ag5.com/mail-assets/mobile/googleplay/pt-br.png",
+	},
+	{
+		tag:                 language.EuropeanPortuguese, // pt-PT
+		appleStoreLink:      "https://cdn.ag5.com/mail-assets/mobile/appstore/pt.png",
+		googlePlayStoreLink: "https://cdn.ag5.com/mail-assets/mobile/googleplay/pt.png",
+	},
+	{
+		tag:                 language.Russian, // ru
+		appleStoreLink:      "https://cdn.ag5.com/mail-assets/mobile/appstore/ru.png",
+		googlePlayStoreLink: "https://cdn.ag5.com/mail-assets/mobile/googleplay/ru.png",
+	},
+	{
+		tag:                 language.Swedish, // sv
+		appleStoreLink:      "https://cdn.ag5.com/mail-assets/mobile/appstore/se.png",
+		googlePlayStoreLink: "https://cdn.ag5.com/mail-assets/mobile/googleplay/sv.png",
+	},
+	{
+		tag:                 language.Thai, // th
+		appleStoreLink:      "https://cdn.ag5.com/mail-assets/mobile/appstore/th.png",
+		googlePlayStoreLink: "https://cdn.ag5.com/mail-assets/mobile/googleplay/th.png",
+	},
+	{
+		tag:                 language.Turkish, // tr
+		appleStoreLink:      "https://cdn.ag5.com/mail-assets/mobile/appstore/tr.png",
+		googlePlayStoreLink: "https://cdn.ag5.com/mail-assets/mobile/googleplay/tr.png",
+	},
+	{
+		tag:                 language.Ukrainian, // uk
+		appleStoreLink:      "https://cdn.ag5.com/mail-assets/mobile/appstore/uk.png",
+		googlePlayStoreLink: "https://cdn.ag5.com/mail-assets/mobile/googleplay/uk.png",
+	},
+	{
+		tag:                 language.Vietnamese, // vi
+		appleStoreLink:      "https://cdn.ag5.com/mail-assets/mobile/appstore/vi.png",
+		googlePlayStoreLink: "https://cdn.ag5.com/mail-assets/mobile/googleplay/vi.png",
+	},
+	{
+		tag:                 language.Chinese, // zh
+		appleStoreLink:      "https://cdn.ag5.com/mail-assets/mobile/appstore/zh.png",
+		googlePlayStoreLink: "https://cdn.ag5.com/mail-assets/mobile/googleplay/zh.png",
+	},
+	{
+		tag:                 language.SimplifiedChinese, // zh-Hans
+		appleStoreLink:      "https://cdn.ag5.com/mail-assets/mobile/appstore/zh.png",
+		googlePlayStoreLink: "https://cdn.ag5.com/mail-assets/mobile/googleplay/zh.png",
+	},
+	{
+		tag:                 language.TraditionalChinese, // zh-Hant
+		appleStoreLink:      "https://cdn.ag5.com/mail-assets/mobile/appstore/zh.png",
+		googlePlayStoreLink: "https://cdn.ag5.com/mail-assets/mobile/googleplay/zh.png",
+	},
+}
```

Comment:
I'm not sure if testing a mapping like this is the way to go (also from a maintainability point of view: if we're going to have the full mapping hardcoded in the codebase like this anyway and we're more sure of this test code than the code in the actual logic, then why not just use this as the main logic and not test it :P).

I'd focus on one basic test for the main part of the URL, and then edge cases like there not being any language tags or the language tags not being in the mapping

Reply:
That test was more a 'run it' kind of thing to see if my algorithm was ok when I didn't know go had maps and I tried to make it all dynamic based on the language! But that became a mess with the exceptions and differences so decided to hard code it afterwards which makes it a duplicate indeed.

Added better tests now, but think it is ok to leave it in for now if someone later wants to make an algorithm for it and wants to check if it still gives the same result.